File: lib/hyper-vr/cs/CoinObject.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class CoinObject : MonoBehaviour {

	private bool active;
	private bool tree;
	private CoinObject parent;
	private int children;

	// Use this for initialization
	void Start () {
		
	}
	
	// Update is called once per frame
	void Update () {
		
	}
	
	public void activate() {
		//Debug.Log("Activate");
		active = true;
		transform.GetComponent<Renderer>().material.SetColor("_Color", Color.green);
	}
	
	public void setTree(CoinObject p, int c) {
		tree = true;
		parent = p;
		children = c;
		//Debug.Log("parent: " + parent);
	}
	
	public void alert() {
		//Debug.Log("Alert");
		if (tree) {
			children--;
			if (children <= 0) {
				activate();
			}
		}
	}

	private void OnCollisionEnter(Collision other) {
		//Debug.Log("collision: " + other);
	}

	private void OnTriggerEnter(Collider other) {
		//Debug.Log("trigger: " + other);
		if (active) {
			if (tree && parent != null) {
				parent.alert();
			}
			Destroy(gameObject);
		}
	}
}

File: lib/hyper-vr/coin_object.js
AFRAME.registerComponent('coin_object', {
  init: function () {
    this.bindMethods();

    this.active = false;
    this.tree = null;
    this.coinParent = null;
    this.numChildren = null;

    this.el.addEventListener('collisions', this.onCollision);
  },

  bindMethods: function () {
    this.activate = this.activate.bind(this);
    this.setTree = this.setTree.bind(this);
    this.onCollision = this.onCollision.bind(this);
  },

  activate: function () {
    this.active = true;
    this.el.setAttribute('material', 'color', 'green');
  },

  setTree: function (p, c) {
    this.tree = true;
    this.coinParent = p;
    this.numChildren = c;
  },

  alert: function () {
    if (this.tree) {
      this.numChildren--;
      if (this.numChildren <= 0) {
        this.activate();
      }
    }
  },

  onCollision: function (e) {
    console.log(e);
    if (this.active) {
      if (this.tree && this.coinParent != null) {
        this.coinParent.alert();
      }
      this.el.parentNode.removeChild(this.el);
    }
  },
});
